Release: Improve Structure of Release Notes

sanssecours opened the pull request:
# Purpose

We now use a structure, similar to the one of the repository, for the release notes. Instead of personal interest (maintainer, developer, user) we now use general technical topics (bindings, plugins, tools, …) as headings. This structure should it make much simpler to decide where to look for a certain new feature or bug fix.

# Checklist

- [x] I checked all commit messages.
- [x] This PR updates the release notes.

sanssecours commented:
I canceled the Travis build job for now, since I assume Armin wants to use Travis to debug issue #2050.

markus2330 commented:
Thank you for putting so much thought into the release notes! Maybe you can avoid "Other News" even more? If you want, you can do a release in Juli (where I am on holidays).

There are two things to consider on your proposed change:

1. you won't make many friends if you create a merge conflict with all currently open PRs. (Your "low priority" suggests that this was not your plan anyway.) So maybe such a change is better suited shortly before the release (or together with the release). After releases all PRs need to be rebased anyway.
2. please also update doc/todo/NEWS.md so that our template is also updated for future releases.

sanssecours commented:
> Maybe you can avoid "Other News" even more?

I added the section “Support” in the latest update and moved the info about the updated Fish completion file there.

> you won't make many friends if you create a merge conflict with all currently open PRs.

That is correct. However, I think the current structure of the release notes is quite bad and justifies the drastic changes. Especially [“Notes for Elektra's Developers”](https://github.com/ElektraInitiative/libelektra/blob/master/doc/news/_preparation_next_release.md#notes-for-elektras-developers) is basically just an unordered list of unrelated changes.

markus2330 commented on doc/todo/NEWS.md:
```diff
@@ -102,33 +113,40 @@ These notes are of interest for people maintaining packages of Elektra:
-The website is generated from the repository, so all information about
-plugins, bindings and tools are always up to date. Furthermore, we changed:
+
+## Infrastructure
```
Infrastructure is here twice?

markus2330 commented on doc/todo/NEWS.md:
```diff
@@ -102,33 +113,40 @@ These notes are of interest for people maintaining packages of Elektra:
 
 
-## Fixes
+## Website
```
Doesn't Website belong to the same category as Documentation?

sanssecours replied:
While the website also provides documentation for Elektra, I think website news can include other items, such as updates to the snippet service too.

markus2330 commented on doc/todo/NEWS.md:
```diff
@@ -48,52 +48,63 @@ You can also read the news [on our website](https://www.libelektra.org/news/0.8.
 
-## Other New Features
-
-We added even more functionality, which could not make it to the highlights:
```
A short beginning sentence after every section would be nice. Imho directly starting with lists does not look so nice.

sanssecours replied:
> A short beginning sentence after every section would be nice.

Since the sentences basically just describes the heading title I think we can remove them.

markus2330 replied:
You can be creative with other sentences :wink:

sanssecours replied:
Honestly, I do not really know what do write below the sections that is not just a description of the section title.

markus2330 commented on doc/todo/NEWS.md:
```diff
@@ -48,52 +48,63 @@ You can also read the news [on our website](https://www.libelektra.org/news/0.8.
-## Other New Features
-
-We added even more functionality, which could not make it to the highlights:
+## Bindings
```
What was the reason to start with Bindings and not Plugins? But I actually do not know what is more important for people reading the release notes. I assumed it is the plugins but maybe that is wrong.

sanssecours replied:
> What was the reason to start with Bindings and not Plugins? 

The main reason was alphabetic order 😊. I changed the arrangement in the latest update.

markus2330 commented on doc/todo/NEWS.md:
```diff
@@ -48,52 +48,63 @@ You can also read the news [on our website](https://www.libelektra.org/news/0.8.
 ### <<HIGHLIGHT2>>
 
 
-## Other New Features
```
I think it had some value to have a list of important new features (for advertisement) because sometimes nobody wanted to write a text for the highlight. Then we still could put the feature here.

sanssecours replied:
You can always add this text if you want highlight something for a specific release, but basically I think we should avoid categories, that overlap with other categories. In my opinion it should be easy to decide where to search for a certain item of interest.

markus2330 replied:
Yes, that is a good point.

My idea was to have the maximum number of categories in the template. (Also to avoid merge conflicts.) And shortly before the release we can merge categories that are hardly used.

The short sentences might help to distinguish where to put something.

sanssecours replied:
> My idea was to have the maximum number of categories in the template. (Also to avoid merge conflicts.) And shortly before the release we can merge categories that are hardly used.

Adding many (non-overlapping) sections for topics is a good idea, but adding sections that are too general does not help in removing merge conflicts. Categories like “Other New Features” or “Notes for Elektra's Developers” apply to more or less every Elektra update. For example, it is not really clear if an new feature of a plugin should be part of “Other New Features” or “Plugins”.

> The short sentences might help to distinguish where to put something.

Could you provide an example for a news item where it is not clear where to put it? I only had problems with the Docker and Jenkins related news items, since I do not know much about both  of those topics.

markus2330 commented on doc/todo/NEWS.md:
```diff
@@ -48,52 +48,63 @@ You can also read the news [on our website](https://www.libelektra.org/news/0.8.
+
 
-We improved the documentation in the following ways:
+## Support
```
What should be written here? (Except of shell completion features, which would also fit to scripts.)

sanssecours replied:
> What should be written here? 

Stuff that does not fit in the other categories? This update was mainly a reaction to your statement from above:

> Maybe you can avoid "Other News" even more?

.

markus2330 commented on doc/news/_preparation_next_release.md:
```diff
@@ -69,175 +69,173 @@ Thanks to Armin Wurzinger.
-These notes are of interest for people maintaining packages of Elektra:
+### CMake
 
+- The build system no longer installs haskell dependencies from hackage by itself, instead
```
haskell -> Haskell

markus2330 commented:
Thank you, great work!
